site/docs/recipes/expressions.md

```diff
@@ -0,0 +1,216 @@
+
+There's also a global function `expressionBuilder` you can use to create expression builders:
+
+```ts
+import { expressionBuilder } from 'kysely'
+
+// `eb1` has type `ExpressionBuilder<DB, 'person'>`
+const eb1 = expressionBuilder<DB, 'person'>()
+
+// In this one you'd have access to tables `person` and `pet` and all their columns.
+const eb2 = expressionBuilder<DB, 'person' | 'pet'>()
+
+let qb = query
+  .selectFrom('person')
+  .innerJoin('movie as m', 'm.director_id', 'person.id')
+
+// You can also provide a query builder instance and the types are inferred automatically.
+// Type of `eb` is `ExpressionBuilder<DB & { m: Movie }, 'person' | 'm'>`
+const eb = expressionBuilder(qb)
+
+qb = qb.where(eb.not(eb.exists(
+  eb.selectFrom('pet')
+    .select('pet.id')
+    .whereRef('pet.name', '=', 'm.name')
+)))
+```
+
+## Creating reusable helpers
+
+The expression builder can be used to create reusable helper functions. Let's say we have a complex `where` expression we want to reuse in multiple queries:
+
+```ts
+function hasDogNamed(name: string): Expression<boolean> {
+  const eb = expressionBuilder<DB, 'person'>()
+
+  return eb.exists(
+    eb.selectFrom('pet')
+      .select('pet.id')
+      .whereRef('pet.owner_id', '=', 'person.id')
+      .where('pet.species', '=', 'dog')
+      .where('pet.name', '=', name)
+  )
+}
+```
+
+This helper can now be used in any query that has the `person` table joined:
+
+```ts
+const doggoPersons = await db
+  .selectFrom('person')
+  .selectAll('person')
+  .where(hasDogNamed('Doggo'))
+  .execute()
+```
+
+The above helper is not very type-safe. The fillowing code would compile, but fail at runtime:
+
+```ts
+const bigFatFailure = await db
+  .selectFrom('movie')
+  .selectAll('movie')
+  .where(hasDogNamed('Doggo'))
+  .execute()
+```
+
+We can write a more type-safe version of the helper like this:
+
+```ts
+function hasDogNamed(name: string){
+  return (eb: ExpressionBuilder<DB, 'person'>) => {
+    return eb.exists(
+      eb.selectFrom('pet')
+        .select('pet.id')
+        .whereRef('pet.owner_id', '=', 'person.id')
+        .where('pet.species', '=', 'dog')
+        .where('pet.name', '=', name)
+    )
+  }
+}
+```
+
+With this helper, you get a type error when trying to use it in a context where the `person` table isn't joined to the query.
+
+## Conditional expressions
+
+In the following, we'll only cover `where` expressions. The same logic applies `having`, `on`, `orderBy`, `groupBy` etc. but not to `select`. If you want to add conditional selections, see [this recipe](https://www.kysely.dev/docs/recipes/conditional-selects).
+
+The most basic and common case for adding conditional `where` expressions is having a set of optional filters you want to combine using `and`. Since the `where`, `having` and other filter functions are additive, most of the time this is enough:
```

Comment:
```suggestion
Having a set of optional filters you want to combine using `and`, is the most basic and common use case of conditional `where` expressions.
Since the `where`, `having` and other filter functions are additive, most of the time this is enough:
```